File: src/pokete/classes/multiplayer/msg/fight/starter.py
from typing import TypedDict
import pokete.bs_rpc as bs_rpc

STARTER_TYPE = "pokete.fight.starter"


class StarterData(TypedDict):
    name: str


class Starter(bs_rpc.Body):
    def __init__(self, data: StarterData):
        super().__init__(STARTER_TYPE, data)

File: pkg/server/pokete/msg/fight/starter.go
package fight

import (
	"github.com/lxgr-linux/pokete/bs_rpc/msg"
)

const StarterType msg.Type = "pokete.fight.starter"

type Starter struct {
	msg.BaseMsg
	Name string `json:"name"`
}

func (s Starter) GetType() msg.Type {
	return StarterType
}

func NewStarter(name string) Starter {
	return Starter{msg.BaseMsg{}, name}
}
